source/common/network/address_impl.cc

```diff
@@ -244,26 +244,50 @@ int Ipv6Instance::socket(SocketType type) const {
   return fd;
 }
 
-PipeInstance::PipeInstance(const sockaddr_un* address) : InstanceBase(Type::Pipe) {
+PipeInstance::PipeInstance(const sockaddr_un* address, socklen_t ss_len)
+    : InstanceBase(Type::Pipe) {
   if (address->sun_path[0] == '\0') {
-    throw EnvoyException("Abstract AF_UNIX sockets not supported.");
+#if !defined(__linux__)
+    throw EnvoyException("Abstract AF_UNIX sockets are only supported on linux.");
+#endif
+    RELEASE_ASSERT(ss_len >= offsetof(struct sockaddr_un, sun_path) + 1);
+    abstract_namespace_ = true;
+    address_length_ = ss_len - offsetof(struct sockaddr_un, sun_path);
   }
   address_ = *address;
-  friendly_name_ = address_.sun_path;
+  friendly_name_ = abstract_namespace_
+                       ? "@" + std::string(address_.sun_path + 1, address_length_ - 1)
```

Comment:
A couple things on this line: 

* please use absl::string_view instead of std::string to avoid an allocation and copy
* Once you're using absl::string_view, I don't know that operator+ is defined for what you want, so do fmt::format("@{}", absl::string_view(address_.sun_path + 1, address_length_ - 1))